napari_skimage_regionprops/_regionprops.py

```diff
@@ -184,3 +185,157 @@ def ellipsoid_axis_lengths(table):
 
 regionprops_table_all_frames = analyze_all_frames(regionprops_table)
 register_function(regionprops_table_all_frames, menu="Measurement > Regionprops of all frames (nsr)")
+
+def make_element_wise_dict(a,b):
+    '''Returns an element-wise dictionary of keys from a and values from b'''
+    return dict(list(map(lambda a, b: [a,b], *[a, b])))
+
+@register_function(menu="Measurement > Regionprops map multichannel (scikit-image, nsr)")
+def regionprops_map_channels_table(image_list : List[napari.types.ImageData], 
+                             labels_list: List[napari.types.LabelsData], 
+                             ref_channel: int = 0,
+                             overlap: float = 0.5, 
+                             summary: bool = True,
+                             size : bool = True, 
+                             intensity : bool = True, 
+                             perimeter : bool = False, 
+                             shape : bool = False, 
+                             position : bool = False,
+                             moments : bool = False, 
+                             napari_viewer : Viewer = None) -> "pandas.DataFrame":
+    
+    current_dim_value = 0
+    if napari_viewer is not None:
+        current_dim_value = napari_viewer.dims.current_step[0]
+
+        # store list of labels layers for saving results later
+        labels_layer_list = []
+        for layer in napari_viewer.layers:
+            print(layer, type(layer), layer.data)
+            if type(layer) is 'napari.layers.labels.labels.Labels':
+                labels_layer_list += [layer]
+        print(labels_layer_list)
+        # to do: deal with 4D data
+        # for image, labels in image_list, labels_list:
+        #     # deal with 4D data
+        #     if len(image.shape) == 4:
+        #         image = image[current_dim_value]
+        #     if len(labels.shape) == 4:
+        #         labels = labels[current_dim_value]
```

Comment:
You might not need to deal with 4D data at all. Try to use this instead:
https://github.com/haesleinhuepf/napari-skimage-regionprops/blob/master/napari_skimage_regionprops/_regionprops.py#L185-L186

Reply:
Thanks! I made changes now to use the decorator, but I still need to test with timelapse data. Maybe timelapse this can wait a second version?